### PR TITLE
Extract factories and add comments to SessionManager

### PR DESCRIPTION
**Pull request by daehn**
# What's in this PR?

* Extract factory classes in separate files.
* Add comments to SessionManager.

### CODE REVIEW

File: Source/SessionManager/SessionManager.swift [hunk 1]
```diff
@@ -23,89 +23,82 @@ import WireUtilities
...
 
-    public typealias LaunchOptions = [UIApplicationLaunchOptionsKey : Any]
+
+/// The `SessionManager` class handles the creation of `ZMUserSession` and `UnauthenticatedSession`
+/// objects, the handover between them as well as account switching.
+///
+/// There are multiple things neccessary in order to store (and switch between) multiple accounts on one device, a couple of them are:
+/// 1. The folder structure in the app sandbox has to be modeled in a way in which files can be associated with a single account.
+/// 2. The login flow should not rely on any persistent state (e.g. no database has to be created on disk before being logged in).
+/// 3. There has to be a persistent layer storing information about accounts and the currently selected / active account.
```
Comment (vytis) [inline Source/SessionManager/SessionManager.swift]: Typo: persistent -> persistence

Comment (daehn) [inline Source/SessionManager/SessionManager.swift]: @vytis fixed!

File: Source/SessionManager/SessionManager.swift [hunk 2]
```diff
@@ -23,89 +23,82 @@ import WireUtilities
...
+/// The wire account database and a couple of other related files are stored in the shared container in a folder named by the accounts
+/// `remoteIdentifier`. All information about different accounts on a device are stored by the `AccountManager` (see the documentation
+/// of that class for more information). The `SessionManager`s main responsibility at the moment is checking whether there is a selected 
+/// `Account` or not, and creating an `UnauthenticatedSession` or `ZMUserSession` accordingly. An `UnauthenticatedSession` is used
+/// to create requests to either log in existing users or to register new users. It uses its own `UnauthenticatedOperationLoop`, 
+/// which is a stripped down version of the regular `ZMOperationLoop`. This unauthenticated operation loop only uses a small subset
+/// of transcoders needed to perform the login / registration (and related phone number verification) requests. For more information
+/// see `UnauthenticatedOperationLoop`.
+///
+/// The result of using an ``UnauthenticatedSession` is retriving a remoteIdentifier of a logged in user, as well as a valid cookie.
```
Comment (typfel) [inline Source/SessionManager/SessionManager.swift]: double "\`"  in \`\`UnauthenticatedSession\`

Comment (vytis) [inline Source/SessionManager/SessionManager.swift]: Also retriving

**Comment by daehn** [comment]
We should start to run spell checking on comments using `Danger` 😜.

**Pull request merged**